riscv/sim.cc

```diff
@@ -149,15 +149,19 @@ sim_t::sim_t(const cfg_t *cfg, bool halted,
   }
 
   //per core attribute
-  int cpu_offset = 0, rc;
+  int cpu_offset = 0, cpu_map_offset=0, rc;
   size_t cpu_idx = 0;
   cpu_offset = fdt_get_offset(fdt, "/cpus");
-  if (cpu_offset < 0)
+  cpu_map_offset = fdt_get_offset(fdt, "/cpus/cpu-map");
+  if (cpu_offset < 0 || cpu_map_offset < 0)
```

Comment:
Doesn't this change the behavior for DTS without the `cpu-map` node? As in, a DTS with `cpu` nodes, but no `cpu-map` node will skip the code below?

Reply:
Sorry for the silly mistake, I have corrected that below.